### DISCUSSION TITLE
how to create order_channels_channel

### DISCUSSION THREAD

**Discussion by jihadkamil** [vendurehq/vendure#1946] [Q&A]
hello, i wanna ask, can we "create order_channels_channel" manually? i was customizing createOrder feature. my feature hitted by grpc instead of using graphql endpoint. unfortunately it caused the "order_channels_channel" not created in the database, maybe it because of the requestContext is not as complete as from graphql. the impact of empty order_channels_channel is client cannot see the order detail because it's unauthorized
(using vendure-1.3.1)

![image](https://user-images.githubusercontent.com/23089263/208800134-64a2952b-372b-417d-85b0-9a3650d12bfc.png)
(cannot see the order detail)

Regards

**Comment by michaelbromley** [comment]
Hi,

> can we "create order_channels_channel" manually

Do you mean you want to create the actual table, or just add a row to it?

It is safe to just manually add a row, and in future you need to modify your grpc endpoint to make sure a channel gets assigned. Maybe the `ctx` object does not have an active channel ID associated with it. For reference, here is the code that determines the channel in the usual GraphQL or REST request pipeline: https://github.com/vendure-ecommerce/vendure/blob/335dfb5b67fbb24c8c654a7bf0afbd9bd84d19f8/packages/core/src/service/helpers/request-context/request-context.service.ts#L94-L95

**Comment by jihadkamil** [comment]
yes i mean add new row to order_channels_channel

**Reply by jihadkamil** [reply]
hi, can we create new row in the order_channels_channel table manually? 
my code return error message=> "column \"createdAt\" of relation \"order_channels_channel\" does not exist"

my code:
> #OrderChannelsChannel: new entity with 2 columns (channelId & orderId)
>  const occ = { channel: defaultChannel,      order: activeOrder }
> await this.connection.getRepository(ctx, OrderChannelsChannel).save(occ)

**Reply by michaelbromley** [reply]
Hi,
You would create a new row in that table by actually associating the order with the channel like this:

```ts
order.channels = [defaultChannel];
await this.connection.getRepository(ctx, Order).save(order);
```

**Reply by jihadkamil** [reply]
thankyou michael for answering. and it solved now
